ast/format.go

```diff
@@ -73,7 +177,9 @@ func (f *formatter) formatPackage(n *Package) {
 			f.writeRune('\n')
 			f.writeRune('\n')
 		}
-		f.writeComment(file.Name)
+		if len(file.Name) > 0 {
+			f.writeComment(file.Name)
+		}
```

Comment:
This fixes the issue we had when formatting a generated AST that doesn't have a file provenance.
Otherwise, it would generate:

```
package main
//

...
```

The related test is below